src/UI/PersonNewsFeedBar.jsx

```diff
@@ -39,26 +65,42 @@ const PreferencesForm = () => {
     const sourcesString = activeSources.join(",");
 
     navigate(`/personalized-news/${sourcesString}`);
+    console.log("handle save");
+    toggleModal();
   };
 
+  if (!isOpen) return null;
```

Comment:
You can achieve this with the `keepMounted` prop on Dialog component like so:

```ts
<Dialog 
  open={isOpen} 
  onClose={toggleModal}
  keepMounted={false} // <- This will unmount the component from the DOM when `isOpen` is false
>
```